GreedyParameterPointSampler.h

```diff
@@ -0,0 +1,734 @@
+/******************************************************************************
+ *
+ * Copyright (c) 2013-2021, Lawrence Livermore National Security, LLC
+ * and other libROM project developers. See the top-level COPYRIGHT
+ * file for details.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR MIT)
+ *
+ *****************************************************************************/
+
+// Description: This class greedily selects parameter points
+//              for the construction of a ROM database. The implemented
+//              greedy algorithm is derived from algorithm 2 of Choi et. al's
+//              paper "Gradient-based constrained optimization using a database
+//              of linear reduced-order models": https://arxiv.org/abs/1506.07849
+//
+//              The greedy algorithm workflow is as follows:
+//              1. Construct the GreedyParameterPointSampler by giving it a
+//                 domain of parameter points.
+//              2. Request a parameter point to sample.
```

Comment:
It is not clear what it means to sample a parameter point. Some more details would help.